flutter_example/lib/main.gen.dart

```diff
@@ -1,40 +1,47 @@
+import 'package:flutter/foundation.dart';
 import 'package:flutter/material.dart';
 
-import 'macros/auto_dispose.dart';
+import 'macros/auto_listenable.dart';
 import 'macros/functional_widget.dart';
 import 'macros/render_accessors.dart';
 
 void main() {
-  runApp(const MyApp());
+  var counterNotifier = ValueNotifier<int>(0);
+  runApp(MyApp(counterNotifier));
 }
 
 @FunctionalWidget(widgetName: 'MyApp')
-Widget _buildApp(BuildContext context,
+Widget _buildApp(BuildContext context, ValueNotifier<int> counter,
     {String? appTitle, String? homePageTitle}) {
   return MaterialApp(
       title: appTitle ?? 'Flutter Demo',
       theme: ThemeData(primarySwatch: Colors.blue),
-      home: MyHomePage(title: homePageTitle ?? 'Flutter Demo Home Page'));
+      home: MyHomePage(counter,
+          title: homePageTitle ?? 'Flutter Demo Home Page'));
 }
 
 class MyHomePage extends StatefulWidget {
-  const MyHomePage({Key? key, required this.title}) : super(key: key);
+  final ValueNotifier<int> counter;
+
+  const MyHomePage(this.counter, {Key? key, required this.title})
+      : super(key: key);
 
   final String title;
 
   @override
   State<MyHomePage> createState() => _MyHomePageState();
 }
 
+@autoListenable
```

Comment:
I guess you could interpret this as "Hey, this state object listens to all Listenables it can find in the associated widget."

My guess is that most people (well, at least me) will be thinking the other way around: Either "here's a Listenable I want to listen to" or "here's a handler that I want to invoke when the listenable fires". Those two seem like more natural entry points / annotation points.

Reply:
Ya, given the widget/state separate this seemed the most intuitive to me (and least boilerplate). We can make other examples too that work a bit differently - and I do want to extend this one to also work for things from the state class (that are pulled off the context, etc). That will end up looking more like what you are talking about I think.